### PR TITLE
For #7352: integrate highlightable browser menu changes

### PR DESCRIPTION
**Pull request by Baron-Severin**
### Pull Request checklist

- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [x] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [x] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features

### After merge
- [ ] **Milestone**: Make sure issues finished by this pull request are added to the [milestone](https://github.com/mozilla-mobile/fenix/milestones) of the version currently in development.

### To download an APK when reviewing a PR:
1. click on Show All Checks,
2. click Details next to "Taskcluster (pull_request)" after it appears and then finishes with a green checkmark,
3. click on the "Fenix - assemble" task, then click "Run Artifacts".
4. the APK links should be on the left side of the screen, named for each CPU architecture

highlights1
highlights2

### CODE REVIEW

**Comment by Baron-Severin** [comment]
Need to look into a problem with this before it can be merged.  The three-dot menu is supposed to be highlighted any time any of its items can be, but this isn't holding true.

STR
- Open new tab
- Navigate to https://onezero.medium.com/irl-ads-are-taking-scary-inspiration-from-social-media-7088e8241beb
- Wait for load

Expected
- In the browser menu, Reader View is highlighted
- The three dot menu is highlighted

Actual
- In the browser menu, Reader View is highlighted
- The three dot menu is **_not_** highlighted

**Comment by NotWoods** [comment]
You need to call `invalidate` on the toolbar menu so it checks the list again (not a great API right now...sorry!). There's a helper class that just calls invalidate in certain conditions.

File: app/src/main/java/org/mozilla/fenix/components/toolbar/DefaultToolbarMenu.kt [hunk 1]
```diff
@@ -277,26 +278,38 @@ class DefaultToolbarMenu(
...
+        highlight = BrowserMenuHighlight.LowPriority(
+            label = context.getString(R.string.browser_menu_read),
+            notificationTint = getColor(context, R.color.whats_new_notification_color)
+        )
     ) { checked ->
         onItemTapped.invoke(ToolbarMenu.Item.ReaderMode(checked))
     }
 
-    private val readerAppearance = BrowserMenuImageText(
+    private val readerAppearance = BrowserMenuHighlightableItem(
```
Comment (NotWoods) [inline app/src/main/java/org/mozilla/fenix/components/toolbar/DefaultToolbarMenu.kt]: Does appearance also need a highlight? It feels redundant.

Comment (Baron-Severin) [reply app/src/main/java/org/mozilla/fenix/components/toolbar/DefaultToolbarMenu.kt]: @apbitner what do you think?  I have no strong feelings either way on this one.

File: app/src/main/java/org/mozilla/fenix/components/toolbar/DefaultToolbarMenu.kt [hunk 2]
```diff
@@ -14,6 +14,7 @@ import mozilla.components.browser.menu.BrowserMenuBuilder
 import mozilla.components.browser.menu.BrowserMenuHighlight
```
Comment (Baron-Severin) [inline app/src/main/java/org/mozilla/fenix/components/toolbar/DefaultToolbarMenu.kt]: (unrelated to this LoC, just putting it here so we can chain comments)

> You need to call `invalidate` on the toolbar menu so it checks the list again (not a great API right now...sorry!). There's a helper class that just calls invalidate in certain conditions.

@NotWoods hmm, I don't know if we can.  We don't have a reference to the view here, and it's a required parameter for invalidate.  I remember going through this problem with Christian a few months ago and being more or less stumped.  Do you have any suggestions?

Comment (NotWoods) [reply app/src/main/java/org/mozilla/fenix/components/toolbar/DefaultToolbarMenu.kt]: You could override `onReaderableStateUpdated` in `MenuPresenter`

Comment (Baron-Severin) [reply app/src/main/java/org/mozilla/fenix/components/toolbar/DefaultToolbarMenu.kt]: Works great, thanks for the help!

**Comment by NotWoods** [comment]
Code looks good, I'll approve once we get a UX decision on the extra appearance dot.

File: app/src/main/java/org/mozilla/fenix/components/toolbar/DefaultToolbarMenu.kt (hunk 1 above)
Comment (apbitner) [reply app/src/main/java/org/mozilla/fenix/components/toolbar/DefaultToolbarMenu.kt]: I don't think it's necessary. 'Appearance' may not live in the toolbar menu permanently anyways.

Comment (Baron-Severin) [reply app/src/main/java/org/mozilla/fenix/components/toolbar/DefaultToolbarMenu.kt]: Thanks Adam, I'll pull it out.

File: app/src/main/java/org/mozilla/fenix/components/toolbar/DefaultToolbarMenu.kt [hunk 3]
```diff
@@ -277,10 +278,14 @@ class DefaultToolbarMenu(
...
+    private val readerMode = BrowserMenuHighlightableSwitch(
         label = context.getString(R.string.browser_menu_read),
-        imageResource = R.drawable.ic_readermode,
-        initialState = readerModeStateProvider
+        startImageResource = R.drawable.ic_readermode,
+        initialState = readerModeStateProvider,
+        highlight = BrowserMenuHighlight.LowPriority(
+            label = context.getString(R.string.browser_menu_read),
+            notificationTint = getColor(context, R.color.whats_new_notification_color)
+        )
```
Comment (NotWoods) [inline app/src/main/java/org/mozilla/fenix/components/toolbar/DefaultToolbarMenu.kt]: `isHighlighted` should also be set. I don't think the menu currently checks visibility.

Comment (Baron-Severin) [reply app/src/main/java/org/mozilla/fenix/components/toolbar/DefaultToolbarMenu.kt]: `isHighlighted` defaults to true, which is what we always want here.  But it's probably not a bad idea to put it in here to be explicit.

Comment (NotWoods) [reply app/src/main/java/org/mozilla/fenix/components/toolbar/DefaultToolbarMenu.kt]: If you include these options in the toolbar list when reader mode is unavailable, then the highlight may always appear.

Comment (Baron-Severin) [reply app/src/main/java/org/mozilla/fenix/components/toolbar/DefaultToolbarMenu.kt]: Oh, I see what you mean.  Thanks for clarifying.

Comment (Baron-Severin) [reply app/src/main/java/org/mozilla/fenix/components/toolbar/DefaultToolbarMenu.kt]: We looked into this and it seems that it does filter is visibility when adding the highlight to the three dot menu, so I'm resolving this comment.

**Pull request merged**